src/pkg/packager/common.go

```diff
@@ -264,14 +286,24 @@ func (p *Packager) loadZarfPkg() error {
 		return fmt.Errorf("unable to handle the provided package path: %w", err)
 	}
 
-	alreadyExtracted := p.cfg.DeployOpts.PackagePath == p.tmp.Base
+	alreadyExtracted := p.cfg.PkgOpts.PackagePath == p.tmp.Base
+
+	if alreadyExtracted && pathsToCheck == nil {
+		paths, err := utils.RecursiveFileList(p.tmp.Base, nil, false)
+		if err != nil {
+			return fmt.Errorf("unable to get a list of files in the package: %w", err)
+		}
+		for _, path := range paths {
+			pathsToCheck = append(pathsToCheck, strings.TrimPrefix(path, p.tmp.Base+string(os.PathSeparator)))
+		}
+	}
```

Comment:
I would prefer if we didn't have this shim here since it is janky - assuming this is because library code could do a partial pull into a directory and Zarf may not know about that?

Reply:
If so it might be fine (for now) but we should remove it quickly

Reply:
Correct, as `packager` only accepts a directory as a source, not a collection of file paths.

Reply:
refactor noted in https://github.com/defenseunicorns/zarf/issues/1943#issuecomment-1676007461

Reply:
fine